### DISCUSSION TITLE
设计器左侧组件如何做成动态的？

### DISCUSSION THREAD

**Discussion by henryzp** [alibaba/formily#3187] [Q&A 2.x] [answered]
业务上考虑这样的场景：

以字段为纬度的，可能会在多个页面中出现，因此需要这样的json schema，不至于代码c v

然后这样的schema保存到数据库中

页面渲染时，从数据库表中读取中动态的配置，渲染到左侧组件的区块（或者新加一个区块）。

求给个思路。。。

**Comment by janryWang** [comment, marked as answer]
自己实现一个ResourceWidget，异步拿到所有资源，然后渲染资源面板

**Reply by henryzp** [reply]
我的疑问是这个可以做到。。

但是`ComponentTreeWidget`这个组件不是也要接收到上面的组件么？要怎么处理呢？

如果只是`ResourceWidget`，我可以实现

**Reply by janryWang** [reply]
Proxy拦截一下

**Reply by henryzp** [reply]
<img width="539" alt="image" src="https://user-images.githubusercontent.com/3849702/173361611-f86d337f-6f05-48bc-b959-28d581bb204b.png">

**Reply by henryzp** [reply]
额，Proxy拦截一下？

能给个伪代码么？

**Reply by janryWang** [reply]
不用Proxy也行，用observable创建components对象，然后动态给里面塞组件
